lib/normalizers/v1.js

```diff
@@ -48,14 +48,18 @@ _.assign(Builders.prototype, {
      * Normalizes inherited v1 auth manifests.
      *
      * @param {Object} entityV1 - A v1 compliant wrapped auth manifest.
+     * @param {?Object} options - The set of options for the current auth cleansing operation.
+     * @param {?Boolean} [options.includeNoauth=false] - When set to true, noauth is set to null.
+     *
      * @returns {Object} - A v1 compliant set of auth helper attributes.
      */
-    auth: function (entityV1) {
+    auth: function (entityV1, options) {
         if (!entityV1) { return; }
         // Handle the noauth/normal case for new and legacy attributes first, in that order
-        if (util.notLegacy(entityV1, 'auth') && entityV1.auth) {
-            return (entityV1.auth.type === 'noauth') ? { type: 'noauth' } : util.sanitizeAuthArray(entityV1);
+        if (util.notLegacy(entityV1, 'auth')) {
+            return util.sanitizeAuthArray(entityV1, options);
         }
+        if (entityV1.currentHelper === null) { return this.options.noDefaults ? undefined : { type: 'noauth' }; }
```

Comment:
As discussed in the discussion, `currentHelper=null` should not be converted to `auth.type=noauth`.